### DISCUSSION TITLE
Multiple base URLs in application config for one rest client

### DISCUSSION THREAD

**Discussion by jacksmom** [quarkusio/quarkus#33065] [Q&A] [answered]
Hi,

I have an application where I have one rest client that communicates with one service through a post method "call toggle". I currently have that rest client configured for one instance of that service:

services.ExampleService/mp-rest/url="url"

Client:
```
@Path("/x")
@RegisterRestClient
public interface ExampleService {
	
	  @POST
	  @Path("/toggle")
	  	String callToggle();
}
````
Injecting and calling the client:
```
@Inject
@RestClient
ExampleService service;


@Operation(hidden = true)
@POST
@Path("/toggle")
@Transactional
@Consumes(MediaType.TEXT_PLAIN)
public CompletionStage<String> toggleService(Integer id) {
	service.callToggle();
}
```

I'd like to be able to communicate with multiple instances of the same service, calling this "call toggle" post method to multiple base urls. 

How do I configure multiple base URL's for this same service in the application config and choose which url to use when I call the client?

[a comment by an automated account is not shown]

**Reply by Sgitario** [reply, marked as answer]
You would need to build [the client programmatically](https://quarkus.io/guides/rest-client-reactive#programmatic-client-creation-with-restclientbuilder) injecting the `baseUri` from the application config. For example:

`application.properties`:
```
my.services=http://server:port/x/toggle,http://server:port/y/toggle
```

`creating the client programmatically`:
```
@ConfigProperty("my.services")
List<String> myServices;

List<ExampleService> services;

public void setup() {
  for (String myService : myServices) {
     services.add(RestClientBuilder.newBuilder()
            .baseUri(URI.create(myService))
            .build(ExampleService.class));
  } 
}
```

**Reply by cescoffier** [reply]
You may be able to use a stork discovery provider. But the API approach is probably better (and simpler)
